modules/local/kraken2/main.nf

```diff
@@ -0,0 +1,39 @@
+process KRAKEN2 {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(reads), path(kraken2_db)
+
+    output:
+    tuple val(meta), path("*_kraken2_output.tsv"), path("*_kraken2_report.txt"), emit:report
+    path "versions.yml",                                                         emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def paired = meta.paired_end ? "--paired" : ""
+
+    """
+    kraken2 \\
+        --db ${kraken2_db} \\
+        --threads $task.cpus \\
+        --output ${meta.id}_kraken2_output.tsv \\
+        --report ${meta.id}_kraken2_report.txt \\
+        --gzip-compressed \\
+        $paired \\
+        $args \\
+        $reads
```

Comment:
Could you gzip the output files here as well? In particular`kraken2_report.txt` can be quite large.

That is, add, after the kraken2 command:

```nextflow
gzip ${meta.id}_kraken2_output.tsv
gzip ${meta.id}_kraken2_report.txt
```

And then you will have to change the `output` expressions to match the gzipped files.

Although, I think Bracken requires the uncompressed report.txt. So maybe leave that one ungzipped.

Reply:
I think Bracken does requires the uncompressed report.txt - but I compressed it for now and made a note of the change. 😄
Fixed in [3a76046](https://github.com/phac-nml/speciesabundance/pull/5/commits/3a7604675b4d2a6063ff6cacc632d9c49f4d8bf8).